api/v1alpha1/retry_types.go

```diff
@@ -18,6 +18,12 @@ type Retry struct {
 	// +kubebuilder:default=2
 	NumRetries *int32 `json:"numRetries,omitempty"`
 
+	// NumRetriesToAcrossPriorities defines the number of requests (initial attempt + retries)
+	// that should be sent to the same priority before switching to a different one.
+	//
+	// +optional
+	NumRetriesToAcrossPriorities *int32 `json:"numRetriesToAcrossPriorities,omitempty"`
```

Comment:
Can we change this to `NumAttemptsPerPriority`? We also need to document that omitting this (i.e. value is `0`) means there is no limit to the number of attempts per priority.

Reply:
1) piggy backing off @cmaddalozzo's suggestion, if there are more setting related to `perPriority`
this could be represented as
```
perPriority:
  numAttemps: 2

```

else `numAttemptsPerPriority` makes sense

Reply:
> numAttemptsPerPriority makes sense

it sounds reasonable. fixed